api/v1/replication-contraints.go

```diff
@@ -0,0 +1,120 @@
+package api
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/data-preservation-programs/spade-tenant/db"
+	"github.com/google/uuid"
+	"github.com/labstack/echo/v4"
+	"gorm.io/gorm"
+)
+
+type ReplicationConstraints []ReplicationConstraint
+type ReplicationConstraint db.ReplicationConstraint
+
+type ReplicationConstraintResponse struct {
+	ConstraintID  db.ID `json:"constraint_id"`
+	ConstraintMax int   `json:"constraint_max"`
+}
+
+func (a *apiV1) ConfigureReplicationConstraintsRouter(e *echo.Group) {
+	g := e.Group("/collections/:collectionUUID/replication-constraints")
+	g.PUT("", a.handleSetReplicationConstraints)
+	g.DELETE("/:replicationConstraintId", a.handleDeleteReplicationConstraints)
+	g.GET("/:replicationConstraintId", a.handleGetReplicationConstraints)
+}
+
+// handleSetReplicationConstraints godoc
+//
+//	@Summary		Creates or updates ReplicationConstraints associated with a tenant and collection
+//	@Param			token header string true "Auth token"
+//	@Param			ReplicationConstraints body
+//	@Produce		json
+//	@Success		200	{object}	ResponseEnvelope{response=ReplicationConstraints}
+//	@Router			/collections/:collectionUUID/replication-constraints [put]
+func (a *apiV1) handleSetReplicationConstraints(c echo.Context) error {
+	var replicationConstraints []ReplicationConstraintResponse
+
+	err := json.NewDecoder(c.Request().Body).Decode(&replicationConstraints)
+
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, CreateErrorResponseEnvelope(c, http.StatusInternalServerError, err.Error()))
+	}
+
+	err = a.db.Transaction(func(tx *gorm.DB) error {
+		for _, replicationConstraint := range replicationConstraints {
+			constraint := &db.ReplicationConstraint{CollectionID: uuid.MustParse(c.Param("collectionUUID")), ConstraintID: replicationConstraint.ConstraintID, ConstraintMax: replicationConstraint.ConstraintMax}
+
+			res := tx.Save(&constraint)
+
+			if res.Error != nil {
+				return err
+			}
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, CreateErrorResponseEnvelope(c, http.StatusInternalServerError, err.Error()))
+	}
+
+	return c.JSON(http.StatusOK, CreateSuccessResponseEnvelope(c, "Successfully created or updated replication constraint"))
+}
+
+// handleDeleteReplicationConstraints godoc
+//
+//	@Summary		Delete ReplicationConstraints used by a tenant
+//	@Param			token header string true "Auth token"
+//	@Produce		json
+//	@Success		200	{object}	ResponseEnvelope{response=ReplicationConstraints}
+//	@Router			/collections/:collectionUUID/replication-constraints/:replicationConstraintId [delete]
```

Comment:
Can you update the comment to make it very clear that it only deletes a single ReplicationConstraint? As the routes here are plural (i.e, constraint**s**) it kinda feels like it should delete multiple of them. But I'm okay with it only deleting a single one as long as its called out